[14.0][FIX] account_statement_import_online: use unittest.mock and don't import tests at install

yajo opened the pull request:
Installing this module failed because `mock` couldn't be found. Indeed there was a migration problem.

However, I was not testing the module; just installing. At install, tests shouldn't be imported.

Fixing both problems here.

@moduon MT-295

rafaelbn commented:
Hello @cormaza @ChrisOForgeFlow what do you think? Thank you! :-)

rafaelbn commented:
Hello @NL66278 @njeudy what do you think? 🙏 Thank you!

ChrisOForgeFlow commented:
Maybe should be good delete account_statement_import_online/tests/online_bank_statement_provider_dummy.py file too

NL66278 commented:
The dummy model is needed for running the tests, as can be seen from all tests failing at the moment. Why not just correct the import of the mock?

yajo commented:
Let me see if I can fix it with https://github.com/OCA/odoo-test-helper.

yajo committed:
[FIX] account_statement_import_online: use unittest.mock and don't import tests at install

Installing this module failed because `mock` couldn't be found. Indeed there was a migration problem.

However, I was not testing the module; just installing. At install, tests shouldn't be imported.

This file was being imported because the dummy model was expected to exist in database for tests. Using odoo-test-helper for that now.

@moduon MT-295

yajo commented:
Let's see how it goes now.

FTR `mock` still has to be installed until https://github.com/OCA/odoo-test-helper/pull/19 is merged, which is a tiny PR. Please review if possible.

yajo commented:
Tests are ✔️ now.

NL66278 reviewed (approved):
:+1: LGTM

pedrobaeza commented:
/ocabot merge patch
